docs/auto-create-collections-by-product-type-or-vendor/script.liquid

```diff
@@ -0,0 +1,347 @@
+                    "name": {{ sales_channel_names[0] | json }}
+                  }
+                }
+              ]
+            }
+          }
+        }
+      {% endcapture %}
+
+      {% assign result = result_json | parse_json %}
+    {% endif %}
+
+    {% assign publications = result.data.publications.edges | map: "node" %}
+    {% assign publication_names = publications | map: "name" | sort %}
+    {% assign publications_indexed_by_name = publications | index_by: "name" %}
+
+    {% for sales_channel_name in sales_channel_names %}
+      {% if publication_names contains sales_channel_name %}
+        {% assign publication_ids = publication_ids | push: publications_indexed_by_name[sales_channel_name].id %}
+
+      {% else %}
+        {% unless event.preview %}
+          {% error
+            message: "A configured sales channel name does not match any of the publication names available in this shop.",
+            sales_channel_name: sales_channel_name,
+            publication_names: publication_names
+          %}          
+        {% endunless %}
+      {% endif %}
+    {% endfor %}
+  {% endif %}
+
+  {% if create_collections_by_product_type %}
+    {% for product_type in product_types %}
+      {% capture query %}
+        query {
+          collections(
+            first: 1
+            query: {{ product_type | json | prepend: "title:" | json }}
+          ) {
+            edges {
+              node {
+                id
+                title
+                handle
+              }
+            }
+          }
+        }
+      {% endcapture %}
+
+      {% assign result = query | shopify %}
+
+      {% if result.data.collections.edges != blank %}
+        {% log
+          message: "A collection title matching this product type already exists; skipping.",
+          product_type: product_type,
+          collection: result.data.collections.edges.first.node
+        %}
+
+      {% else %}
+        {% capture mutation %}
+          mutation {
+            collectionCreate(
+              input: {
+                title: {{ product_type | json }}
+                ruleSet: {
+                  appliedDisjunctively: false
+                  rules: [
+                    {
+                      column: TYPE
+                      relation: EQUALS
+                      condition: {{ product_type | json }}
+                    }
+                  ]
+                }
+              }
+            ) {
+              collection {
+                id
+                title
+                handle
+              }
+              userErrors {
+                field
+                message
+              }
+            }
+          }
+        {% endcapture %}
+
+        {
+          "action": {
+            "type": "shopify",
+            "options": {{ mutation | json }},
+            "meta": {
+              "publication_ids": {{ publication_ids | json }}
+            }
+          }
+        }
```

Comment:
Can you phrase this (and the one below) with an argumentless `{% action %}` tag, por favor? Nice use of action metadata, by the way. :D 👏